src/coreclr/jit/jitmetadata.cpp

```diff
@@ -0,0 +1,124 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "jitpch.h"
+#include "jitmetadata.h"
+
+#ifdef DEBUG
+
+//------------------------------------------------------------------------
+// JitMetadata::getName: Get the name corresponding to a JitMetadataName value,
+// which can be used to report metadata back to the EE.
+//
+// Parameters:
+//   name - The JitMetadataName enum entry
+//
+// Returns:
+//   String representation of the name
+//
+const char* JitMetadata::getName(JitMetadataName name)
+{
+    switch (name)
+    {
+#define JITMETADATA(name, type, flags)                                                                                 \
+    case JitMetadataName::name:                                                                                        \
+        return #name;
+#include "jitmetadatalist.h"
+
+        default:
+            unreached();
+    }
+}
+
+//------------------------------------------------------------------------
+// JitMetadata::report: Report metadata back to the EE.
+//
+// Parameters:
+//   comp - Compiler instance
+//   name - Enum entry specifying the metadata name
+//   data - Pointer to the value to report back
+//
+void JitMetadata::report(Compiler* comp, JitMetadataName name, const void* data)
+{
+    comp->info.compCompHnd->reportMetadata(getName(name), data);
```

Comment:
Should the `report` functions do nothing if `RunningSuperPmiReplay()` is false? (I guess the VM does nothing anyway)

Reply:
Yeah I figured that we should just let it be up to the EE. If the EE wants to do something with the information then that's up to them. (Also, we don't check for SPMI in release JITs)